src/os/interaction/modifyinteraction.js

```diff
@@ -409,6 +410,21 @@ class Modify extends OLModify {
     if (this.clone_) {
       const originalGeom = this.clone_.getGeometry();
       if (originalGeom) {
+        // While handling events, OpenLayers may add a new Z (altitude) coordinate to a set of XY
+        // coordinates. Since it is not intended for XY coordinates to have a Z coordinate, these
+        // new values can potentially be undefined and cause problems with Cesium. To prevent this,
+        // set all undefined values to zero.
+        if (originalGeom instanceof Polygon) {
```

Comment:
This is currently called each time the geometry is updated, which happens frequently during the modify interaction. We should at most need to call it once, because once the altitude values are fixed they will no longer be undefined.

We could do that when we clone the feature and add altitude values to the coordinate array, knowing OL will change the geometry layout to XYZ. However, the reason that happens is the event coordinate has a Z component in 3D mode which conflicts with OL's assumption that the event coordinate layout will match the modify geometry's layout. We can fix that in our `handleDownEvent_` override by checking if `evt.coordinate.length > stride` and if so, set the length to the stride to crop the coordinate. We will also want to override the `handleDragEvent_` function to make the same change. Between those, we can ensure the event coordinate matches the geometry we're modifying, and we won't have to change the geometry's layout at all.